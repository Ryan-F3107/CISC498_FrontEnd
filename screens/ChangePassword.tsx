import React from 'react';
import { Text, View, TouchableOpacity, Platform, KeyboardAvoidingView } from 'react-native';
import { IconButton, TextInput, Divider } from 'react-native-paper';
import { showMessage } from 'react-native-flash-message';
import styles from '../styling/styles';
import { ScrollView } from 'react-native-gesture-handler';
import { serverAddress } from './connectToBackend';

class ChangePassword extends React.Component {

    constructor(props) {
        super(props);
        const initalState = {
            oldPassword: '',
            newPassword: '',
            confirmNewPassword: '',

            errorOldPassword: '',
            errorNewPassword: '',
            errorConfirmNewPassword: ''
        }
        this.state = initalState;
    }


    // Verify that all the required fields are filled in
    checkForm() {
        let decision = false;

        if (this.state.oldPassword == "" || this.state.newPassword.length < 8 || this.state.confirmNewPassword != this.state.newPassword) {
            decision = false
        }
        else {
            decision = true
        }
        return decision

    }

    render() {
        return (
            <KeyboardAvoidingView
                behavior={Platform.OS === "ios" ? "padding" : "height"}
                style={{ flex: 1 }}
            >
                <View style={styles.homeContainer}>

                    {/*Close Button*/}
                    <IconButton
                        style={styles.closeButton}
                        icon="close"
                        size={35}
                        color={'black'}
                        onPress={() => {
                            this.setState({ oldPassword: '', newPassword: '', confirmNewPassword: '' });

                            if (this.props.route.params.accountType == "customer") {
                                this.props.navigation.goBack();
                                this.setState({
                                    oldPassword: '',
                                    newPassword: '',
                                    confirmNewPassword: '',
                                    errorOldPassword: '',
                                    errorNewPassword: '',
                                    errorConfirmNewPassword: ''
                                })
                            } else if (this.props.route.params.accountType == "business") {
                                this.props.navigation.goBack();
                            }
                        }}
                    ></IconButton>

                    <Text
                        style={{
                            fontSize: 30,
                            paddingHorizontal: Platform.OS === "ios" ? 70 : 50
                        }}
                    >
                        Change Password
                    </Text>

                    <ScrollView>

                        <View style={{
                            flex: 1,
                            paddingHorizontal: Platform.OS === "ios" ? 70 : 50,
                            backgroundColor: '#fafafa'
                        }}>


                            <View style={{ marginTop: 30 }}
                            />
                            <Text
                                style={{ marginTop: 5, marginBottom: -5 }}
                            > * Please enter your current password {"\n"} to verify your identity.
                            </Text>

                            {/*Current Password. This is to verify whether the user is the right user or not.*/}
                            <TextInput
                                style={styles.textInputPassword}
                                label="CURRENT PASSWORD"
                                mode="outlined"
                                placeholder="Enter current password"
                                autoCapitalize='none'
                                spellCheck={false}
                                secureTextEntry={true}
                                theme={{ colors: { primary: '#0a0540' } }}
                                onChangeText={oldPassword => this.setState({ oldPassword: oldPassword })}
                                value={this.state.oldPassword}
                                onBlur={() => { // If the field is left blank, show an error message 
                                    if (this.state.oldPassword == "") {
                                        this.setState({ errorOldPassword: "Required" });
                                    }
                                }}
                                onFocus={() => { // When the field is tapped, remove the error message
                                    this.setState({ errorOldPassword: "" });
                                }}
                            />
                            <Text style={styles.errorMessage}>{this.state.errorOldPassword}</Text>

                            <Divider style={{ borderBottomWidth: 1.5, borderColor: 'grey', width: 300, alignSelf: 'center', marginTop: 30, marginBottom: 15 }} />

                            {/*New Password. Must be at least 8 characters long*/}
                            <TextInput
                                style={styles.textInputPassword}
                                label="NEW PASSWORD"
                                mode="outlined"
                                placeholder="Enter new password"
                                autoCapitalize='none'
                                secureTextEntry={true}
                                theme={{ colors: { primary: '#0a0540' } }}
                                onChangeText={newPassword => this.setState({ newPassword: newPassword })}
                                value={this.state.newPassword}
                                onBlur={() => { // If the field is left blank, or has an invalid password, show an error message 
                                    if (this.state.newPassword == "") {
                                        this.setState({ errorNewPassword: "Required", confirmNewPassword: '' });
                                    } else if (this.state.newPassword.length < 8) {
                                        this.setState({ errorNewPassword: "Must be at least 8 characters long" });
                                    }
                                }}
                                onFocus={() => { // When the field is tapped, remove the error message
                                    this.setState({ errorNewPassword: "" });
                                }}
                            />
                            <Text style={styles.errorMessage}>{this.state.errorNewPassword}</Text>

                            {/*Re-enter the password*/}
                            <TextInput
                                style={styles.textInputConfirmPassword}
                                label="CONFIRM PASSWORD"
                                mode="outlined"
                                autoCapitalize='none'
                                secureTextEntry={true}
                                theme={{ colors: { primary: '#0a0540' } }}
                                disabled={this.state.newPassword.length < 8}
                                onChangeText={confirmNewPassword => this.setState({ confirmNewPassword: confirmNewPassword })}
                                value={this.state.confirmNewPassword}
                                onBlur={() => { // If the field is left blank or if the new password & the re-entered password don't match, show an error message 
                                    if (this.state.confirmNewPassword == "") {
                                        this.setState({ errorConfirmPassword: "Required" });
                                    } else if (this.state.newPassword != this.state.confirmNewPassword) {
                                        this.setState({ errorConfirmNewPassword: "Paswords do not match" });
                                    }
                                }}
                                onFocus={() => { // When the field is tapped, remove the error message
                                    this.setState({ errorConfirmNewPassword: "" })
                                }}
                            />
                            <Text style={styles.errorMessage}>{this.state.errorConfirmNewPassword}</Text>

                            {/*Place the button correctly based on the platform*/}
                            <View style={{
                                marginBottom: 50
                            }} />
                            {/*When the "Confirm" button is clicked, 
                            send the new & confirmed password to the backend, where it sends back a code to notify whether the change was successful or not*/}
                            <TouchableOpacity
                                style={styles.button}
                                onPress={async () => {

                                    if (this.checkForm()) {

                                        var link = `${serverAddress}/checkin/change_password/` + this.props.route.params.receivedUserInfo["id"] + "/";

                                        let response = await fetch(link, {
                                            method: 'PUT',
                                            headers: {
                                                Authorization: 'Bearer ' + this.props.route.params.receivedUserInfo["access"],
                                                Accept: 'application/json',
                                                'Content-Type': 'application/json'
                                            },
                                            body: JSON.stringify({

                                                old_password: this.state.oldPassword,
                                                new_password: this.state.confirmNewPassword
                                            })
                                        }
                                        )
                                        response = await response.status;

                                        // Password has been successfully changed
                                        // Display the flash message at the top
                                        if (response == 200) {
                                            showMessage({
                                                message: "Password changed!",
                                                type: "success",
                                                autoHide: true,
                                                duration: 2000,
                                                backgroundColor: "#0a0540",
                                                color: "#fafafa",
                                                icon: "success"
                                            });
                                            this.setState({ oldPassword: '', newPassword: '', confirmNewPassword: '' });
                                            this.props.navigation.goBack();

                                        } else { // Error status code from the backend
                                            showMessage({
                                                message: `Error: Password could not be changed. ${'\n'}${'\n'}Please double-check your current password and try again.`,
                                                type: "danger",
                                                autoHide: true,
                                                duration: 2500,
                                                backgroundColor: "#ff504a",
                                                color: "#fafafa",
                                                icon: "danger"
                                            });
                                        }
                                    } else { // Textfield is incomplete
                                        showMessage({
                                            message: `Error: Incomplete. ${'\n'}${'\n'}Please fill in all the fields.`,
                                            type: "danger",
                                            autoHide: true,
                                            duration: 2500,
                                            backgroundColor: "#ff504a",
                                            color: "#fafafa",
                                            icon: "danger"
                                        });
                                    }

                                }}
                            >

                                <Text style={{ color: '#fafafa', alignSelf: 'center' }}>Confirm</Text>

                            </TouchableOpacity>


                        </View>
                    </ScrollView>
                </View>


            </KeyboardAvoidingView>




        )
    }
}

export default ChangePassword;