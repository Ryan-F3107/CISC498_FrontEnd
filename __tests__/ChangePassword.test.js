import React from 'react';
import renderer from 'react-test-renderer';
import ChangePassword from '../screens/ChangePassword';

describe("<ChangePassword />", () => {
    it("has 1 child", () => {
        const tree = renderer.create(<ChangePassword />).toJSON();
        expect(tree.children.length).toBe(1);
    });

    it('renders correctly', () => {
        renderer.create(<ChangePassword />);
    });

    it('ChangePassword testing with snapshot', () => {
        const tree = renderer.create(<ChangePassword />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});