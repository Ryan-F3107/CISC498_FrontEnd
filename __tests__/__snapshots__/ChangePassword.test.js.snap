// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChangePassword /> ChangePassword testing with snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#fafafa",
        "flex": 1,
        "paddingBottom": 20,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": undefined,
        }
      }
      accessible={true}
      focusable={true}
      hitSlop={
        Object {
          "bottom": 6,
          "left": 6,
          "right": 6,
          "top": 6,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "margin": 6,
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 26.25,
              "height": 52.5,
              "width": 52.5,
            },
            undefined,
            Object {
              "alignSelf": "flex-end",
              "marginRight": 10,
              "marginTop": 40,
            },
          ],
        ]
      }
    >
      <View>
        <Text />
      </View>
    </View>
    <Text
      style={
        Object {
          "fontSize": 30,
          "paddingHorizontal": 70,
        }
      }
    >
      Change Password
    </Text>
    <RCTScrollView
      collapsable={false}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
    >
      <View>
        <View
          style={
            Object {
              "backgroundColor": "#fafafa",
              "flex": 1,
              "paddingHorizontal": 70,
            }
          }
        >
          <View
            style={
              Object {
                "marginTop": 30,
              }
            }
          />
          <Text
            style={
              Object {
                "marginBottom": -5,
                "marginTop": 5,
              }
            }
          >
             * Please enter your current password 
            

             to verify your identity.
          </Text>
          <View
            style={
              Object {
                "paddingTop": 16,
              }
            }
          >
            <View>
              <View
                pointerEvents="none"
                style={
                  Array [
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 6,
                    },
                    Object {
                      "backgroundColor": "#fafafa",
                      "borderColor": "rgba(0, 0, 0, 0.54)",
                      "borderRadius": 4,
                      "borderWidth": 1,
                    },
                  ]
                }
              />
              <View
                style={
                  Object {
                    "minHeight": 64,
                    "paddingBottom": 0,
                    "paddingTop": 8,
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                      ],
                    }
                  }
                >
                  <View
                    pointerEvents="none"
                    style={
                      Object {
                        "backgroundColor": "#fafafa",
                        "bottom": 0,
                        "height": 2,
                        "left": 10,
                        "opacity": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 6,
                        "transform": Array [
                          Object {
                            "translateX": 0,
                          },
                        ],
                        "width": 8,
                      }
                    }
                  />
                  <Text
                    numberOfLines={1}
                    style={
                      Object {
                        "backgroundColor": "#fafafa",
                        "color": "transparent",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "left": 18,
                        "opacity": 0,
                        "paddingHorizontal": 0,
                        "position": "absolute",
                        "textAlign": "left",
                        "top": 37,
                        "transform": Array [
                          Object {
                            "translateX": 0,
                          },
                          Object {
                            "translateY": 0,
                          },
                          Object {
                            "scale": 1,
                          },
                          Object {
                            "scaleY": 1,
                          },
                        ],
                        "writingDirection": "ltr",
                      }
                    }
                  >
                    CURRENT PASSWORD
                  </Text>
                  <Text
                    numberOfLines={1}
                    onLayout={[Function]}
                    style={
                      Object {
                        "color": "#0a0540",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "left": 0,
                        "opacity": 0,
                        "paddingHorizontal": 14,
                        "position": "absolute",
                        "textAlign": "left",
                        "top": 36,
                        "transform": Array [
                          Object {
                            "translateX": 0,
                          },
                          Object {
                            "translateY": 0,
                          },
                          Object {
                            "scale": 1,
                          },
                        ],
                        "writingDirection": "ltr",
                      }
                    }
                  >
                    CURRENT PASSWORD
                  </Text>
                  <Text
                    numberOfLines={1}
                    style={
                      Object {
                        "color": "rgba(0, 0, 0, 0.54)",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "left": 0,
                        "opacity": 0,
                        "paddingHorizontal": 14,
                        "position": "absolute",
                        "textAlign": "left",
                        "top": 36,
                        "transform": Array [
                          Object {
                            "translateX": 0,
                          },
                          Object {
                            "translateY": 0,
                          },
                          Object {
                            "scale": 1,
                          },
                        ],
                        "writingDirection": "ltr",
                      }
                    }
                  >
                    CURRENT PASSWORD
                  </Text>
                </View>
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="none"
                  editable={true}
                  multiline={false}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  placeholder=""
                  placeholderTextColor="rgba(0, 0, 0, 0.54)"
                  rejectResponderTermination={true}
                  secureTextEntry={true}
                  selectionColor="#0a0540"
                  spellCheck={false}
                  style={
                    Array [
                      Object {
                        "flexGrow": 1,
                        "margin": 0,
                        "paddingHorizontal": 14,
                        "zIndex": 1,
                      },
                      Object {
                        "height": 56,
                      },
                      Object {
                        "paddingBottom": 0,
                        "paddingTop": 0,
                      },
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "textAlign": "left",
                        "textAlignVertical": "center",
                      },
                      Array [
                        Object {},
                      ],
                    ]
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
          </View>
          <Text
            style={
              Object {
                "color": "red",
                "fontSize": 11,
                "paddingBottom": 0,
              }
            }
          >
            
          </Text>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(0, 0, 0, 0.12)",
                  "height": 0.5,
                },
                undefined,
                Object {
                  "alignSelf": "center",
                  "borderBottomWidth": 1.5,
                  "borderColor": "grey",
                  "marginBottom": 15,
                  "marginTop": 30,
                  "width": 300,
                },
              ]
            }
          />
          <View
            style={
              Object {
                "paddingTop": 16,
              }
            }
          >
            <View>
              <View
                pointerEvents="none"
                style={
                  Array [
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 6,
                    },
                    Object {
                      "backgroundColor": "#fafafa",
                      "borderColor": "rgba(0, 0, 0, 0.54)",
                      "borderRadius": 4,
                      "borderWidth": 1,
                    },
                  ]
                }
              />
              <View
                style={
                  Object {
                    "minHeight": 64,
                    "paddingBottom": 0,
                    "paddingTop": 8,
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                      ],
                    }
                  }
                >
                  <View
                    pointerEvents="none"
                    style={
                      Object {
                        "backgroundColor": "#fafafa",
                        "bottom": 0,
                        "height": 2,
                        "left": 10,
                        "opacity": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 6,
                        "transform": Array [
                          Object {
                            "translateX": 0,
                          },
                        ],
                        "width": 8,
                      }
                    }
                  />
                  <Text
                    numberOfLines={1}
                    style={
                      Object {
                        "backgroundColor": "#fafafa",
                        "color": "transparent",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "left": 18,
                        "opacity": 0,
                        "paddingHorizontal": 0,
                        "position": "absolute",
                        "textAlign": "left",
                        "top": 37,
                        "transform": Array [
                          Object {
                            "translateX": 0,
                          },
                          Object {
                            "translateY": 0,
                          },
                          Object {
                            "scale": 1,
                          },
                          Object {
                            "scaleY": 1,
                          },
                        ],
                        "writingDirection": "ltr",
                      }
                    }
                  >
                    NEW PASSWORD
                  </Text>
                  <Text
                    numberOfLines={1}
                    onLayout={[Function]}
                    style={
                      Object {
                        "color": "#0a0540",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "left": 0,
                        "opacity": 0,
                        "paddingHorizontal": 14,
                        "position": "absolute",
                        "textAlign": "left",
                        "top": 36,
                        "transform": Array [
                          Object {
                            "translateX": 0,
                          },
                          Object {
                            "translateY": 0,
                          },
                          Object {
                            "scale": 1,
                          },
                        ],
                        "writingDirection": "ltr",
                      }
                    }
                  >
                    NEW PASSWORD
                  </Text>
                  <Text
                    numberOfLines={1}
                    style={
                      Object {
                        "color": "rgba(0, 0, 0, 0.54)",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "left": 0,
                        "opacity": 0,
                        "paddingHorizontal": 14,
                        "position": "absolute",
                        "textAlign": "left",
                        "top": 36,
                        "transform": Array [
                          Object {
                            "translateX": 0,
                          },
                          Object {
                            "translateY": 0,
                          },
                          Object {
                            "scale": 1,
                          },
                        ],
                        "writingDirection": "ltr",
                      }
                    }
                  >
                    NEW PASSWORD
                  </Text>
                </View>
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="none"
                  editable={true}
                  multiline={false}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  placeholder=""
                  placeholderTextColor="rgba(0, 0, 0, 0.54)"
                  rejectResponderTermination={true}
                  secureTextEntry={true}
                  selectionColor="#0a0540"
                  style={
                    Array [
                      Object {
                        "flexGrow": 1,
                        "margin": 0,
                        "paddingHorizontal": 14,
                        "zIndex": 1,
                      },
                      Object {
                        "height": 56,
                      },
                      Object {
                        "paddingBottom": 0,
                        "paddingTop": 0,
                      },
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "textAlign": "left",
                        "textAlignVertical": "center",
                      },
                      Array [
                        Object {},
                      ],
                    ]
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
          </View>
          <Text
            style={
              Object {
                "color": "red",
                "fontSize": 11,
                "paddingBottom": 0,
              }
            }
          >
            
          </Text>
          <View
            style={
              Object {
                "paddingTop": 12,
              }
            }
          >
            <View>
              <View
                pointerEvents="none"
                style={
                  Array [
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 6,
                    },
                    Object {
                      "backgroundColor": "#fafafa",
                      "borderColor": "rgba(0, 0, 0, 0.26)",
                      "borderRadius": 4,
                      "borderWidth": 1,
                    },
                  ]
                }
              />
              <View
                style={
                  Object {
                    "minHeight": 64,
                    "paddingBottom": 0,
                    "paddingTop": 8,
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                      ],
                    }
                  }
                >
                  <View
                    pointerEvents="none"
                    style={
                      Object {
                        "backgroundColor": "#fafafa",
                        "bottom": 0,
                        "height": 2,
                        "left": 10,
                        "opacity": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 6,
                        "transform": Array [
                          Object {
                            "translateX": 0,
                          },
                        ],
                        "width": 8,
                      }
                    }
                  />
                  <Text
                    numberOfLines={1}
                    style={
                      Object {
                        "backgroundColor": "#fafafa",
                        "color": "transparent",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "left": 18,
                        "opacity": 0,
                        "paddingHorizontal": 0,
                        "position": "absolute",
                        "textAlign": "left",
                        "top": 37,
                        "transform": Array [
                          Object {
                            "translateX": 0,
                          },
                          Object {
                            "translateY": 0,
                          },
                          Object {
                            "scale": 1,
                          },
                          Object {
                            "scaleY": 1,
                          },
                        ],
                        "writingDirection": "ltr",
                      }
                    }
                  >
                    CONFIRM PASSWORD
                  </Text>
                  <Text
                    numberOfLines={1}
                    onLayout={[Function]}
                    style={
                      Object {
                        "color": "rgba(0, 0, 0, 0.54)",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "left": 0,
                        "opacity": 0,
                        "paddingHorizontal": 14,
                        "position": "absolute",
                        "textAlign": "left",
                        "top": 36,
                        "transform": Array [
                          Object {
                            "translateX": 0,
                          },
                          Object {
                            "translateY": 0,
                          },
                          Object {
                            "scale": 1,
                          },
                        ],
                        "writingDirection": "ltr",
                      }
                    }
                  >
                    CONFIRM PASSWORD
                  </Text>
                  <Text
                    numberOfLines={1}
                    style={
                      Object {
                        "color": "rgba(0, 0, 0, 0.26)",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "left": 0,
                        "opacity": 0,
                        "paddingHorizontal": 14,
                        "position": "absolute",
                        "textAlign": "left",
                        "top": 36,
                        "transform": Array [
                          Object {
                            "translateX": 0,
                          },
                          Object {
                            "translateY": 0,
                          },
                          Object {
                            "scale": 1,
                          },
                        ],
                        "writingDirection": "ltr",
                      }
                    }
                  >
                    CONFIRM PASSWORD
                  </Text>
                </View>
                <TextInput
                  allowFontScaling={true}
                  autoCapitalize="none"
                  editable={false}
                  multiline={false}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  placeholder=""
                  placeholderTextColor="rgba(0, 0, 0, 0.26)"
                  rejectResponderTermination={true}
                  secureTextEntry={true}
                  selectionColor="rgba(0, 0, 0, 0.54)"
                  style={
                    Array [
                      Object {
                        "flexGrow": 1,
                        "margin": 0,
                        "paddingHorizontal": 14,
                        "zIndex": 1,
                      },
                      Object {
                        "height": 56,
                      },
                      Object {
                        "paddingBottom": 0,
                        "paddingTop": 0,
                      },
                      Object {
                        "color": "rgba(0, 0, 0, 0.54)",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "textAlign": "left",
                        "textAlignVertical": "center",
                      },
                      Array [
                        Object {},
                      ],
                    ]
                  }
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
          </View>
          <Text
            style={
              Object {
                "color": "red",
                "fontSize": 11,
                "paddingBottom": 0,
              }
            }
          >
            
          </Text>
          <View
            style={
              Object {
                "marginBottom": 50,
              }
            }
          />
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignSelf": "center",
                "backgroundColor": "#0a0540",
                "borderRadius": 25,
                "opacity": 1,
                "padding": 15,
                "width": 120,
              }
            }
          >
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "color": "#fafafa",
                }
              }
            >
              Confirm
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
